sympy/functions/elementary/miscellaneous.py

```diff
@@ -265,11 +265,11 @@ def root(arg, n, k=0):
     """
     n = sympify(n)
     if k:
-        return Pow(arg, S.One/n)*S.NegativeOne**(2*k/n)
-    return Pow(arg, 1/n)
+        return Pow(arg, S.One/n, evaluate=evaluate)*S.NegativeOne**(2*k/n)
```

Comment:
The multiplication `*` will evaluate when k is not zero. Use 
```
Mul(Pow(arg, S.One/n, evaluate=evaluate), S.NegativeOne**(2*k/n), evaluate=evaluate)
```
to prevent this. 
Add a test like `root(16, 4, 2, evaluate=False).has(Pow)` to assert that evaluation of Pow is not happening.

Reply:
@normalhuman , I get two errors on performing this step.
```
>>> assert root(x, n, k) == (-1)**(2*k/n)*x**(1/n)
AssertionError
>>> assert real_root(-8, 3) == -2
AssertionError
```
This was not present before I introduced Mul. I can't possibly fathom the source of this error.
What should I do?

Reply:
This is because `AssocOp.__new__` does not handle evaluate flag correctly (which Pow does). If evaluate is None, it should default to `global_evaluate[0]`, but it just keeps the value None. Proposed solution:

```diff
--- a/sympy/core/operations.py
+++ b/sympy/core/operations.py
@@ -31,7 +31,10 @@ def __new__(cls, *args, **options):
         args = list(map(_sympify, args))
         args = [a for a in args if a is not cls.identity]
 
-        if not options.pop('evaluate', global_evaluate[0]):
+        evaluate = options.get('evaluate')
+        if evaluate is None:
+            evaluate = global_evaluate[0]
+        if not evaluate:
             return cls._from_args(args)
```

This would be easier if instead of  `**options` we just had `evaluate=None` (like in `Pow.__new__` but this can't be done in Python 2 because of `*args`.